java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -1095,8 +1116,8 @@ public ExplodedGraphWalkerFactory(List<JavaFileScanner> scanners) {
       seChecks.addAll(checks);
     }
 
-    public ExplodedGraphWalker createWalker(BehaviorCache behaviorCache) {
-      return new ExplodedGraphWalker(alwaysTrueOrFalseChecker, seChecks, behaviorCache);
+    public ExplodedGraphWalker createWalker(BehaviorCache behaviorCache, SemanticModel semanticModel) {
```

Comment:
Wondering if we wouldn't want to realy on a Function<String, Type> to pass a lambda to the factory rather than introducing a coupling with SemanticModel.